src/widget/weffectmetaknob.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "effects/defs.h"
+#include "widget/wknobcomposed.h"
+
+class EffectsManager;
+
+// This is used for effect parameter knobs with dynamic
+// tooltips, if the knob value is displayed by rotating a single SVG..
+class WEffectMetaKnob : public WKnobComposed {
+    Q_OBJECT
+  public:
+    WEffectMetaKnob(QWidget* pParent, EffectsManager* pEffectsManager);
+
+    void setup(const QDomNode& node, const SkinContext& context) override;
+
+  private slots:
+    void effectChanged();
+
+  private:
+    EffectsManager* m_pEffectsManager;
+    EffectSlotPointer m_pEffectSlot;
```

Comment:
encode constraints and ownership in the type system (`gls::not_null<EffectsManager*>` would blow the scope too much right now). implicit info should at least receive a comment.
```suggestion
    gls::not_null<std::weak_ptr<EffectsManager>> m_pEffectsManager;
    // m_pEffectSlot can be null, in which case the effect does not have an effect assigned.
    EffectSlotPointer m_pEffectSlot;
```

Reply:
Ahmm, and change it everywhere in src/effects ??

There is `VERIFY_OR_DEBUG_ASSERT(m_pEffectSlot)` in
`WEffectMetaKnob::setup` and
`WEffectParameterKnobComposed::setup`
and in case of null the respective slots where `m_pEffectSlot`/`m_pParameterSlot` are accessed are not connected.

Reply:
> Ahmm, and change it everywhere in src/effects

In the long term, yes. Not in this PR.

> There is VERIFY_OR_DEBUG_ASSERT(m_pEffectSlot)

Yes, but only on setup. The raw pointer could become dangling, thus a `std::weak_ptr` to ensure that can't happen (and to prevent ref cycles, which would likely form if it was replaced by `std::shared_ptr`. 

Don't get me wrong, the current solution works, no doubts, it's just not very robust IMO.